nbconvert/nbconvertapp.py

```diff
@@ -398,18 +423,21 @@ def convert_notebooks(self):
                 """
             )
             self.exit(1)
-        
+
         # initialize the exporter
         self.exporter = exporter_map[self.export_format](config=self.config)
 
         # no notebooks to convert!
-        if len(self.notebooks) == 0:
+        if len(self.notebooks) == 0 and not self.from_stdin:
             self.print_help()
             sys.exit(-1)
 
         # convert each notebook
-        for notebook_filename in self.notebooks:
-            self.convert_single_notebook(notebook_filename)
+        if not self.from_stdin:
+            for notebook_filename in self.notebooks:
+                self.convert_single_notebook(notebook_filename)
+        else:
+            self.convert_single_notebook(None, input_buffer=sys.stdin)
```

Comment:
Related to comment about l404: 

`self.convert_single_notebook('notebook', input_buffer=sys.stdin)`